docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,143 @@
+
+By using the block supports mechanism via `block.json`, the block author is able to create the same experience than before just by writing a couple of lines.
+
+Besides the obvious advantage of having to do less work to achieve the same results, there's a few other advantages:
+
+- the style information of the block becomes available for the native mobile apps and in the server
+- the block will use the UI controls other blocks use for the same styles, creating a more coherent user experience
+- the UI controls in use by the block will be automatically updated as they are improved, without the block author having to do anything
+
+#### Current limits of the Block Supports API
+
+While the Block Supports API provides value, it also comes with some limitations a block author needs to be aware of. To better visualize what they are, let's run with the following example of a table block:
+
+```html
+<table>
+  <thead><tr><th>Header</th></tr></thead>
+  <tbody>
+    <tr><th>First</th></tr>
+    <tr><th>Second</th></tr>
+  </tbody>
+  <tfoot>
+    <tr><th>Footer</th></tr>
+  </tfoot>
+</table>
+```
+
+1. **Only one style type per block.**
+
+One of the limitations is that, from all the [styles available](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/), there can be only one instance of any them in use by the block. Following the example, the table block can only have a single font size. If the block author wanted to have three different font sizes (head, body, and footer) it can't do it using the current block supports API. See [this issue](https://github.com/WordPress/gutenberg/issues/33255) for more detailed info and ways forward.
+
+2. **Styles are serialized to the outermost HTML node of the block, the wrapper.**
+
+The Block Supports API only serializes the font size value to the wrapper, resulting in the following HTML `<table class="has-small-font-size">` . The current block supports API doesn't serialize this value to a different node, for example, the `<tbody>`.
+
+This is an active area of work you can follow [in the tracking issue](https://github.com/WordPress/gutenberg/issues/38167). The linked proposal is exploring a different way to serialize the user changes: instead of each block support serializing its own data (for example, classes such as `has-small-font-size`, `has-green-color`) the idea is the block would get a single class instead (for example, `wp-style-UUID`) and the contents of that class will be generated in the server by WordPress.
+
+While work continues in that proposal, there's a scape hatch, an experimental option block authors can use. Any block support can skip the serialization to HTML markup by using  `__experimentalSkipSerialization`. For example:
```

Comment:
> Any block support can skip the serialization to HTML markup by using `__experimentalSkipSerialization`.

I think it is worth warning block authors here of further limitations when skipping serialization of block support styles. 

When skipping serialization, normally so certain styles can be applied to inner elements, you often run into the problem where the styles generated via theme.json will not target that inner element and the elements API isn't yet sufficient to be a solution either. 

This is most highlighted by a use case where a block author needs multiple block support toolsets, e.g. spacing (especially padding), typography, and color, with one set applied on the main block wrapper and another targeting an inner element. In such a case, the block author cannot rely on the single selector available within the block.json configuration to match multiple targets. 

An illustration of the [above use case is the Search block](https://github.com/WordPress/gutenberg/issues/32417).

Reply:
That's a good point. I'm working on the global styles doc and will revisit this after to see what we can do.

Reply:
Added a note about this at "current limits of global styles > Can't target different parts of the block". Going to resolve this. Please, bring any improvements to that in the own section itself.